src/test/datascience/widgets/standard.vscode.common.test.ts

```diff
@@ -33,488 +33,642 @@ import { WidgetRenderingTimeoutForTests } from './constants';
 import { getTextOutputValue } from '../../../kernels/execution/helpers';
 
 /* eslint-disable @typescript-eslint/no-explicit-any, no-invalid-this */
-suite('IPyWisdget Tests', function () {
-    const templateRootPath: Uri =
-        workspace.workspaceFolders && workspace.workspaceFolders.length > 0
-            ? urlPath.joinPath(workspace.workspaceFolders[0].uri, 'widgets', 'notebooks')
-            : Uri.file('');
-    let api: IExtensionTestApi;
-    const disposables: IDisposable[] = [];
-    let vscodeNotebook: IVSCodeNotebook;
-    let kernelProvider: IKernelProvider;
-
-    this.timeout(120_000);
-    let previousWidgetScriptSourcesSettingValue: string[] | undefined = undefined;
-    const widgetScriptSourcesValue = ['jsdelivr.com', 'unpkg.com'];
-    // Retry at least once, because ipywidgets can be flaky (network, comms, etc).
-    this.retries(1);
-    suiteSetup(async function () {
-        traceInfo('Suite Setup VS Code Notebook - Execution');
+[false].forEach((useCDN) => {
+    /**
+     * Testing the following permutations:
+     * - VSCode Web + Remote Jupyter + CDN
+     * - VSCode Web + Remote Jupyter + No CDN
+     * - Desktop + Remote Jupyter + CDN
+     * - Desktop + Remote Jupyter + No CDN
+     * - Desktop + Local Kernel + CDN
+     * - Desktop + Local Kernel + No CDN
```

Comment:
Ensure we run the tests in more permutations (previously the tests would only run with CDN, now they'll run with & without CDN in all environments - web, desktop & remote)